sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -737,8 +750,22 @@ private Mono<CosmosBatchResponse> executeBatchRequest(PartitionKeyRangeServerBat
             }
         }
 
-        return this.docClientWrapper.executeBatchRequest(
-            BridgeInternal.getLink(this.container), serverRequest, options, false);
+        return withContext(context -> {
+            final Mono<CosmosBatchResponse> responseMono = this.docClientWrapper.executeBatchRequest(
+                BridgeInternal.getLink(this.container), serverRequest, options, false);
+
+            return BridgeInternal.getTracerProvider(this.cosmosClient)
```

Comment:
why we put the traceProvider here? seems like for other operations they all exists in CosmosAsyncContainer

Reply:
This was a bug - bulk operation did not trace - adding this here fixed that gap